src/backend/mpi/coherence.cpp

```diff
@@ -0,0 +1,167 @@
+/**
+ * @file
+ * @brief This file implements selective coherence mechanisms for ArgoDSM
+ * @copyright Eta Scale AB. Licensed under the Eta Scale Open Source License. See the LICENSE file for details.
+ */
+
+#include "../backend.hpp"
+#include "swdsm.h"
+#include "write_buffer.hpp"
+
+// EXTERNAL VARIABLES FROM BACKEND
+extern control_data *cacheControl;
+extern unsigned long *globalSharers;
+extern pthread_mutex_t cachemutex;
+extern pthread_mutex_t wbmutex;
+extern sem_t ibsem;
+extern void * startAddr;
+extern char * barwindowsused;
+extern MPI_Win *globalDataWindow;
+extern MPI_Win sharerWindow;
+extern argo_statistics stats;
+extern argo_byte *touchedcache;
+extern MPI_Comm workcomm;
+extern WriteBuffer<unsigned long>* write_buffer;
+
+/** @brief The size of an ArgoDSM page */
+static const std::size_t pagesize = 4096;
+
+namespace argo {
+	namespace backend {
+		void selective_acquire(void *addr, size_t size){
+			if(size == 0){
+				// Nothing to invalidate
+				return;
+			}
+
+			const std::size_t block_size = pagesize*CACHELINE;
+			const std::size_t page_padding = reinterpret_cast<unsigned long>(addr)%block_size;
```

Comment:
maybe `page_misalignment` captures it better than `page_padding`?

Reply:
Changed as suggested.